BridgeAppSDK/SBAKeychainWrapper.m

```diff
@@ -0,0 +1,317 @@
+            }
+        } else {
+            returnValue = [NSData dataWithData:(__bridge NSData *)data];
+            if (data) {
+                CFRelease(data);
+            }
+        }
+    }
+    return returnValue;
+}
+
+- (BOOL)setData:(NSData *)data
+         forKey:(NSString *)key
+        service:(NSString *)service
+    accessGroup:(NSString *)accessGroup
+          error:(NSError **)error {
+    BOOL returnValue = YES;
+    if (key) {
+        if (!service) {
+            service = SBAKeychainWrapperDefaultService();
+        }
+        
+        NSMutableDictionary *query = [[NSMutableDictionary alloc] init];
+        [query setObject:(__bridge id)kSecClassGenericPassword forKey:(__bridge id)kSecClass];
+        [query setObject:service forKey:(__bridge id)kSecAttrService];
+        [query setObject:key forKey:(__bridge id)kSecAttrAccount];
+#if !TARGET_IPHONE_SIMULATOR && defined(__IPHONE_OS_VERSION_MIN_REQUIRED)
+        if (accessGroup) {
+            [query setObject:accessGroup forKey:(__bridge id)kSecAttrAccessGroup];
+        }
+#endif
+        
+        OSStatus status = SecItemCopyMatching((__bridge CFDictionaryRef)query, NULL);
+        if (status == errSecSuccess) {
+            if (data) {
+                NSMutableDictionary *attributesToUpdate = [[NSMutableDictionary alloc] init];
+                [attributesToUpdate setObject:data forKey:(__bridge id)kSecValueData];
+                
+                status = SecItemUpdate((__bridge CFDictionaryRef)query, (__bridge CFDictionaryRef)attributesToUpdate);
+                if (status != errSecSuccess) {
+                    if (error) {
+                        *error = [NSError errorWithDomain:NSOSStatusErrorDomain
+                                                     code:status
+                                                 userInfo:@{NSLocalizedDescriptionKey: [Localization localizedString:@"KEYCHAIN_UPDATE_ERROR_MESSAGE"]}];
+                    }
+                    returnValue = NO;
+                }
+            } else {
+                [self removeItemForKey:key service:service accessGroup:accessGroup error:error];
+            }
+        } else if (status == errSecItemNotFound) {
+            NSMutableDictionary *attributes = [[NSMutableDictionary alloc] init];
+            [attributes setObject:(__bridge id)kSecClassGenericPassword forKey:(__bridge id)kSecClass];
+            [attributes setObject:service forKey:(__bridge id)kSecAttrService];
+            [attributes setObject:key forKey:(__bridge id)kSecAttrAccount];
+#if TARGET_OS_IPHONE || (defined(MAC_OS_X_VERSION_10_9) && MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_9)
+            [attributes setObject:(__bridge id)kSecAttrAccessibleAfterFirstUnlock forKey:(__bridge id)kSecAttrAccessible];
+#endif
+            [attributes setObject:data forKey:(__bridge id)kSecValueData];
```

Comment:
I'm assuming this is copypasted from RK, so if this is actually a bug it's a bug in RK as well, but... if data is nil, in the case where the item already existed, it would just remove it with no further comment, but if it didn't already exist, it will try to add the item with nil data, and presumably end up failing with a message that it encountered an error trying to add the item. Just double-checking that this is the desired behavior in both cases, or at least, that maintaining parity with the RK behavior is more of a concern.